### ISSUE TITLE
MoonInfo question on Daylight Darkness panel

### ISSUE THREAD

**Issue by Leosirth** [steepleian/weewx-Weather34#4]
Hi,

Looking on the implementation of the Daylight Darkness panel in **index.php**:

```
<div class=weather-item><div class=chartforecast>
<?php if ($purpleairhardware=='yes'){echo''?>
  <span class="yearpopup">
<a alt="moonphase" title="moonphase" href=mooninfo.php data-lity><?php echo $chartinfo?> <?php echo $lang['Moon'],"Info";}?></a></span>  
<span class="yearpopup"><a alt="meteor showers" title="meteor showers" href="meteorshowers.php" data-lity><?php echo $meteorinfo;?> &nbsp;<?php if ($meteor_default=='No Meteor') {echo "Meteor Showers";} else {	echo $meteor_default;}?></a></span>
<span class="yearpopup"><a alt="aurora information" title="aurora information" href=aurora.php data-lity><?php echo $info;?> Aurora <?php if ($kp>=5) {echo '<oorange>Active</oorange>';}else {echo "";}?></a></span>


</div>
<span class='moduletitle'><?php echo $lang['Daylight']. " | ". $lang['Darkness'];?></span><br />
  <div id="moonphase"></div> </div></div></div>
```

I do not get why **$purpleairhardware** should be enabled in order to show the **MoonInfo** footer.

Additionally trying to load the **MoonInfo** I get an error for variable **$months,$days,$years** not defined.

Is this also happening to anyone else?

I can't see those being defined anywhere in the code.
Do you know what should be the values?

Thanks and regards,

**Comment by balloonhotair** [comment]
Hi,
How are trying to enable Mooninfo? I do not have purpleair and my moon info is just fine. The values are defined in weather34skydata.php

**Comment by steepleian** [comment]
The question whether Purpleair is present or not only relates to the popup window links at the bottom of the module. You get different choices depending on the answer to that that question.
Ian

**Comment by Leosirth** [comment]
@steepleian Correct, the question is why the **purpleairhardware** needs to be present in order to show the mooninfo popup? How are these related?

@balloonhotair Do you have the Mooninfo popup? I'm talking about this:

![image](https://user-images.githubusercontent.com/800911/106249506-b9bc7080-6212-11eb-90a3-a7abb5cbed87.png)

**Comment by steepleian** [comment]
Francesco, The simple answer is that it does not have to be present. The possible combinations and permutations are endless so the selection is historical based on user preferences. You can add any link you wish to the bottom of any module (subject to physical space). Have a look inside index.php and you will see how the popup link works. Its very easy for some one of your skills to edit. If there is something which is a more logical approach, we can discuss it. This is very much a collaborative project. Ian   Sent from Mail for Windows 10 From: FrancescoSent: 29 January 2021 08:17To: steepleian/weewx-Weather34Cc: Ian Millard; MentionSubject: Re: [steepleian/weewx-Weather34] MoonInfo question on Daylight Darkness panel (#4) @steepleian Correct, the question is why the purpleairhardware needs to be present in order to show the mooninfo popup? How are these related?@balloonhotair Do you have the Mooninfo popup? I'm talking about this:—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by Leosirth** [comment]
Hi Ian,

yes I would like to fix this, but I need to understand the logic behind this.
If there is a valid reason to hide the mooninfo popup for missing **purpleairhardware**.

In my opinion this is wrong since the mooninfo has nothing to do with the purpleair sensors, but before changing it I wanted to check with you if there were an hidden logic.

Maybe this was removed due to malfunctions with a dirty patch.

**Comment by steepleian** [comment]
Hi Francesco,

I inherited this code about 2 years ago from the original designer after I showed how it could be ported to be used as a WeeWX skin/template. There have been several issues along the way way which we have had to resolve, be it coding or functionality. Along with my co-collaborator Jerry, we have moved things from PHP into Python where it is sensible and more efficient to do so. I can’t take much credit from this part of the exercise as Jerry is the real Python expert. Things buried in the code will continue to surprise, of that I am sure. As previously stated, this is a collaborative project, listen to the users, what do they want, is it practical, implement. Of course Jerry and I have our own ideas as well but at the end of the day, we are users too. So if you as a user see things that you can improve and will enhance thats fine by me. Just look at the logs and you find the code riddled with non fatal errors, but it still works.

So back to your original question, as far as I know, there is no hidden logic that I am aware of. It just happened to be that way and nobody has ever challenged it.

Ian
